### PR TITLE
feat(client): unity feature flags

### LINKED ISSUE #6575: Integrate feature flag changes in Prisma Client
Integrate the changes introduced in this PR: https://github.com/prisma/prisma-engines/pull/1849

### PR DESCRIPTION
**Pull request by williamluke4**
closes https://github.com/prisma/prisma/issues/6575
[Engines PR](https://github.com/prisma/prisma-engines/pull/1849)

### CODE REVIEW

File: src/packages/engine-core/src/NAPIEngine.ts [hunk 1]
```diff
@@ -41,6 +41,7 @@ type QueryEngineConfig = {
   datasourceOverrides?: Record<string, string>
   logLevel: QueryEngineLogLevel
   telemetry?: QueryEngineTelemetry
+  configDir?: string
```
Comment (Jolg42) [inline src/packages/engine-core/src/NAPIEngine.ts:44]: Actually, it seems it's unused? Can't find anything related to `configDir`. So we can remove?

File: src/packages/sdk/src/engineCommands.ts [hunk 2]
```diff
@@ -75,42 +75,15 @@ export async function getDMMF({
...
-        )
-    }
 
-    const experimentalFlags =
-      enableExperimental &&
-      Array.isArray(enableExperimental) &&
-      enableExperimental.length > 0
-        ? [`--enable-experimental=${enableExperimental.join(',')}`]
-        : []
+    enableExperimental?.forEach((f) => {
```
Comment (Jolg42) [inline src/packages/sdk/src/engineCommands.ts:79]: Would be great to rename `enableExperimental` everywhere to `previewFeatures` maybe? this old naming is confusing I think?

File: src/packages/sdk/src/engineCommands.ts [hunk 3]
```diff
@@ -75,42 +75,15 @@ export async function getDMMF({
...
+      const removedMessage = removedFeatureFlagMap[f]
+      if (removedMessage && !process.env.PRISMA_HIDE_PREVIEW_FLAG_WARNINGS) {
+        console.log(removedMessage)
```
Comment (Jolg42) [inline src/packages/sdk/src/engineCommands.ts:82]:
```suggestion
        console.warn(removedMessage)
```

**Review by Jolg42 — DISMISSED** [review]
Nice 🧹!

See minor comments

File: src/packages/engine-core/src/NAPIEngine.ts (hunk 1 above)
Comment (williamluke4) [reply src/packages/engine-core/src/NAPIEngine.ts:44]: L269

Comment (williamluke4) [reply src/packages/engine-core/src/NAPIEngine.ts:44]: It is definitely used and needed

Comment (Jolg42) [reply src/packages/engine-core/src/NAPIEngine.ts:44]: I saw this indeed `configDir: this.config.cwd,` but I couldn't find where it's actually used 🤔

Comment (Jolg42) [reply src/packages/engine-core/src/NAPIEngine.ts:44]: Seems unused, passed as extra

Comment (williamluke4) [reply src/packages/engine-core/src/NAPIEngine.ts:44]: Some [context](https://github.com/prisma/prisma-engines/blob/master/query-engine/query-engine-napi/src/engine.rs#L96) from the Rust side

Comment (Jolg42) [reply src/packages/engine-core/src/NAPIEngine.ts:44]: It's actually required in engine, thanks Will for the link https://github.com/prisma/prisma-engines/blob/master/query-engine/query-engine-napi/src/engine.rs

**Review by Jolg42 — APPROVED** [review]
👍

**Pull request merged**